src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -142,7 +141,31 @@ protected void messageReceivedDecorate(
             }
 
             // bypass non-netty requests
-            if (channelType.equals("direct")) {
+            if (getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER) != null
```

Comment:
This logic in the receiver was simplified using @krishna-ggk 's suggestion here: https://github.com/opensearch-project/security/pull/2765#discussion_r1192128758

The receiver does not need any logic to determine where the request came from, if the transient headers are present then there is no need to deserialize. If the serialized (non-transient) headers are present then it will deserialize them.